include/nbl/asset/IDescriptorSetLayout.h

```diff
@@ -78,15 +81,68 @@ enum E_DESCRIPTOR_TYPE : uint32_t
 template<typename SamplerType>
 class NBL_API IDescriptorSetLayout : public virtual core::IReferenceCounted
 {
+	struct SBindingRedirect
+	{
+		static constexpr inline uint32_t Invalid = ~0u;
+
+		SBindingRedirect() : bindings(nullptr), offsets(nullptr), count(0ull) {}
+
+		SBindingRedirect(const size_t _count) : count(_count)
+		{
+			bindings = std::make_unique<uint32_t[]>(count << 1);
+			offsets = bindings.get() + count;
+		}
+
+		// For a given binding number `binding` in `bindings`, return its offset in `offsets`.
+		inline uint32_t operator[](const uint32_t binding) const
+		{
+			const uint32_t index = searchForBinding(binding);
+			if (index == Invalid)
+				return Invalid;
+
+			return offsets[index];
+		}
+
+		// Returns index into the `bindings` and `offsets` array for the given binding number `binding.
+		// Assumes `bindings` is sorted.
```

Comment:
and that there are no duplicate values in the array

Reply:
If the user knows what they are doing, I don't know how can we ever have duplicate binding numbers?

Reply:
its good to spell it out in the comments

Reply:
Resolved.